HPCC-24510 Ubuntu 20.04 dependency change to python2/python3 explicit…

Michael-Gardner opened the pull request:
… packages

Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

Commit:
HPCC-24510 Ubuntu 20.04 dependency change to python2/python3 explicit packages

Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

HPCCSmoketest commented:
Process of PR-14042, label: HPCC-24510 starts now.
The reason of this test is: Forced to re-test.
Commit ID: b7245b1c918aed74666783b4c776ae1fea99431d
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-205.ca-central-1.compute.internal

Michael-Gardner commented:
@xwang2713 please test and review.

I tested by:

1) removing all python packages that I could from my 20.04 box and installing the 'python' virtual package.   That includes python2 and python-is-python2.

2) removed everything again, and installed the python3 and python-is-python3 packages.

3) removed everything and installed the hpccsystems-platform package and did an apt install -f to fix dependencies.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-205.ca-central-1.compute.internal
Sha: b7245b1c918aed74666783b4c776ae1fea99431d
Build: success
Milestone:Install hpccsystems-platform-community_7.10.9-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |23 sec  (00:00:23) |
|test (hthor) |883 |883 |0 |315 sec  (00:05:15) |
|test (thor) |821 |821 |0 |777 sec  (00:12:57) |
|test (roxie) |962 |962 |0 |407 sec  (00:06:47) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  763 sec (00:12:43) |  119 sec (00:01:59) |   23 sec (00:00:23) |   19 sec (00:00:19) | 1851 sec (00:30:51) |   19 sec (00:00:19) | 2804 sec (00:46:44) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: b7245b1c918aed74666783b4c776ae1fea99431d
Build: success
Milestone:Install hpccsystems-platform-community_7.10.11-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |25 sec  (00:00:25) |
|test (hthor) |883 |883 |0 |595 sec  (00:09:55) |
|test (thor) |821 |821 |0 |1041 sec  (00:17:21) |
|test (roxie) |962 |962 |0 |725 sec  (00:12:05) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1062 sec (00:17:42) |  125 sec (00:02:05) |   27 sec (00:00:27) |   18 sec (00:00:18) | 2719 sec (00:45:19) |   18 sec (00:00:18) | 3978 sec (01:06:18) |

xwang2713 reviewed (commented):
Looks good

richardkchapman commented:
@ghalliday Does this fix your issues?

ghalliday commented:
@richardkchapman I think this will require python2 and python3 to be installed.  That's better than the previous situation, but we should possibly drop the requirement for python2 at some point.

richardkchapman commented:
That point is 7.12, I think

ghalliday commented:
Yes.  I'll merge.  We also need to make sure the regression suite engine no longer requires python2 for that release.
